### ISSUE TITLE
Question about custom training

### ISSUE THREAD

**Issue by smjk2124** [ultralytics/ultralytics#3395]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, there was a post or two about this, but I wanted to get another opinion on how one would go about this (an example use case, as listed below):

I wish to train YOLOv8 on a dataset for multi-class detection. However, I want to make some changes to training. Other than just detecting object instances (class and bbox location), I also want to inject other auxiliary information (let's take time of day, label 0 for day and label 1 for night) and train the model to predict this correctly. Hence, the changes would be:

- Edit the dataloading process to also return a label as mentioned above for time of day.
- Edit the forward pass of the setup to incorporate support for classifying the time of day as 0/1.
- Edit the loss function (add another weighted term such as BCE) to also account for the time of day classification.

I would train the whole setup starting from pretrained checkpoints (for the bbox detector) with my new dataset, also incorporating the additional classification requirement.

I just wanted to get your thoughts on how to go about this and which files would be the best to edit to incorporate such a change since you're much more familiar with the repository and would be aware of the optimal way to achieve this. Thanks a ton!

**Comment by glenn-jocher** [comment]
@smjk2124 hello! 

Great question! To achieve this, you would need to make changes to the dataloading process, forward pass, and loss function as you mentioned.

To incorporate changes to the dataloading process, you would need to edit the corresponding data loading script for your custom dataset. Here you can include the time of day information.

Next, you can make the necessary changes to the forward pass, where you can incorporate support for the time of day classification as an additional output.

Finally, you can add another weighted loss term such as Binary Cross Entropy (BCE) to account for the time of day classification in the loss function.

You can then train the whole setup starting from pretrained checkpoints as you suggested.

In terms of which files to edit, you can refer to the data loading script, setup functions, and loss functions in the codebase.

I hope this helps! Let me know if you have any further questions.
